### ISSUE TITLE
Unable to install Langflow on Windows w/ Python 3.13

### ISSUE THREAD

**Issue by cpolanish** [langflow-ai/langflow#4474]
### Bug Description

I've been trying to install Langflow on my Windows machine and been running into a bunch of dependency errors. I've managed to work through most of them, but this one I'm stuck on. I'm running Python 3.13 (3.12 was the most recent version in the dropdown when filling out the bug report). Here is the full trace:

<details>
<summary>Install log</summary>

```
cpolanish@FBS-DevAVD-CP2  ~  python -m pip install langflow -U
Collecting langflow
  Using cached langflow-0.0.78.tar.gz (1.4 MB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting beautifulsoup4<5.0.0,>=4.11.2 (from langflow)
  Using cached beautifulsoup4-4.12.3-py3-none-any.whl.metadata (3.8 kB)
Collecting chromadb<0.4.0,>=0.3.21 (from langflow)
  Using cached chromadb-0.3.29-py3-none-any.whl.metadata (6.9 kB)
Collecting dill<0.4.0,>=0.3.6 (from langflow)
  Using cached dill-0.3.9-py3-none-any.whl.metadata (10 kB)
Collecting docstring-parser<0.16,>=0.15 (from langflow)
  Using cached docstring_parser-0.15-py3-none-any.whl.metadata (2.4 kB)
Collecting fake-useragent<2.0.0,>=1.1.3 (from langflow)
  Using cached fake_useragent-1.5.1-py3-none-any.whl.metadata (15 kB)
Collecting fastapi<0.93.0,>=0.92.0 (from langflow)
  Using cached fastapi-0.92.0-py3-none-any.whl.metadata (25 kB)
Collecting google-api-python-client<3.0.0,>=2.79.0 (from langflow)
  Using cached google_api_python_client-2.151.0-py2.py3-none-any.whl.metadata (6.7 kB)
Collecting google-search-results<3.0.0,>=2.4.1 (from langflow)
  Using cached google_search_results-2.4.2.tar.gz (18 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting gptcache<0.2.0,>=0.1.23 (from langflow)
  Using cached gptcache-0.1.44-py3-none-any.whl.metadata (24 kB)
Collecting gunicorn<21.0.0,>=20.1.0 (from langflow)
  Using cached gunicorn-20.1.0-py3-none-any.whl.metadata (3.8 kB)
Collecting huggingface-hub<0.14.0,>=0.13.3 (from langflow)
  Using cached huggingface_hub-0.13.4-py3-none-any.whl.metadata (7.5 kB)
Collecting langchain<0.0.177,>=0.0.176 (from langflow)
  Using cached langchain-0.0.176-py3-none-any.whl.metadata (11 kB)
Collecting llama-cpp-python==0.1.50 (from langflow)
  Using cached llama_cpp_python-0.1.50.tar.gz (1.2 MB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting lxml<5.0.0,>=4.9.2 (from langflow)
  Using cached lxml-4.9.4.tar.gz (3.6 MB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting networkx<4.0,>=3.1 (from langflow)
  Using cached networkx-3.4.2-py3-none-any.whl.metadata (6.3 kB)
Collecting openai<0.28.0,>=0.27.2 (from langflow)
  Using cached openai-0.27.10-py3-none-any.whl.metadata (13 kB)
Collecting pandas<2.0.0,>=1.5.3 (from langflow)
  Using cached pandas-1.5.3.tar.gz (5.2 MB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting psycopg2-binary<3.0.0,>=2.9.6 (from langflow)
  Using cached psycopg2-binary-2.9.10.tar.gz (385 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting pyarrow<12.0.0,>=11.0.0 (from langflow)
  Using cached pyarrow-11.0.0.tar.gz (1.0 MB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting pypdf<4.0.0,>=3.7.1 (from langflow)
  Using cached pypdf-3.17.4-py3-none-any.whl.metadata (7.5 kB)
Collecting pysrt<2.0.0,>=1.1.2 (from langflow)
  Using cached pysrt-1.1.2.tar.gz (104 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting rich<14.0.0,>=13.3.3 (from langflow)
  Using cached rich-13.9.4-py3-none-any.whl.metadata (18 kB)
Collecting tiktoken<0.4.0,>=0.3.3 (from langflow)
  Using cached tiktoken-0.3.3.tar.gz (25 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting typer<0.8.0,>=0.7.0 (from langflow)
  Using cached typer-0.7.0-py3-none-any.whl.metadata (17 kB)
Collecting types-pyyaml<7.0.0.0,>=6.0.12.8 (from langflow)
  Using cached types_PyYAML-6.0.12.20240917-py3-none-any.whl.metadata (1.6 kB)
Collecting unstructured<0.6.0,>=0.5.11 (from langflow)
  Using cached unstructured-0.5.13-py3-none-any.whl.metadata (18 kB)
Collecting uvicorn<0.21.0,>=0.20.0 (from langflow)
  Using cached uvicorn-0.20.0-py3-none-any.whl.metadata (6.2 kB)
Collecting websockets<12.0.0,>=11.0.2 (from langflow)
  Using cached websockets-11.0.3-py3-none-any.whl.metadata (6.6 kB)
Collecting wikipedia<2.0.0,>=1.4.0 (from langflow)
  Using cached wikipedia-1.4.0.tar.gz (27 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting typing-extensions>=4.5.0 (from llama-cpp-python==0.1.50->langflow)
  Using cached typing_extensions-4.12.2-py3-none-any.whl.metadata (3.0 kB)
Collecting soupsieve>1.2 (from beautifulsoup4<5.0.0,>=4.11.2->langflow)
  Using cached soupsieve-2.6-py3-none-any.whl.metadata (4.6 kB)
Collecting requests>=2.28 (from chromadb<0.4.0,>=0.3.21->langflow)
  Using cached requests-2.32.3-py3-none-any.whl.metadata (4.6 kB)
Collecting pydantic<2.0,>=1.9 (from chromadb<0.4.0,>=0.3.21->langflow)
  Using cached pydantic-1.10.19-py3-none-any.whl.metadata (152 kB)
Collecting hnswlib>=0.7 (from chromadb<0.4.0,>=0.3.21->langflow)
  Using cached hnswlib-0.8.0.tar.gz (36 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting clickhouse-connect>=0.5.7 (from chromadb<0.4.0,>=0.3.21->langflow)
  Using cached clickhouse_connect-0.8.6-cp313-cp313-win_amd64.whl.metadata (3.2 kB)
Collecting duckdb>=0.7.1 (from chromadb<0.4.0,>=0.3.21->langflow)
  Using cached duckdb-1.1.3-cp313-cp313-win_amd64.whl.metadata (781 bytes)
INFO: pip is looking at multiple versions of chromadb to determine which version is compatible with other requirements. This could take a while.
Collecting chromadb<0.4.0,>=0.3.21 (from langflow)
  Using cached chromadb-0.3.27-py3-none-any.whl.metadata (6.8 kB)
Collecting pydantic==1.9 (from chromadb<0.4.0,>=0.3.21->langflow)
  Using cached pydantic-1.9.0-py3-none-any.whl.metadata (121 kB)
Collecting chromadb<0.4.0,>=0.3.21 (from langflow)
  Using cached chromadb-0.3.26-py3-none-any.whl.metadata (6.8 kB)
Collecting pydantic>=1.9 (from chromadb<0.4.0,>=0.3.21->langflow)
  Using cached pydantic-2.9.2-py3-none-any.whl.metadata (149 kB)
Collecting numpy>=1.21.6 (from chromadb<0.4.0,>=0.3.21->langflow)
  Using cached numpy-2.1.3-cp313-cp313-win_amd64.whl.metadata (60 kB)
Collecting posthog>=2.4.0 (from chromadb<0.4.0,>=0.3.21->langflow)
  Using cached posthog-3.7.0-py2.py3-none-any.whl.metadata (2.0 kB)
Collecting chromadb<0.4.0,>=0.3.21 (from langflow)
  Using cached chromadb-0.3.25-py3-none-any.whl.metadata (6.7 kB)
Collecting onnxruntime>=1.14.1 (from chromadb<0.4.0,>=0.3.21->langflow)
  Using cached onnxruntime-1.20.0-cp313-cp313-win_amd64.whl.metadata (4.6 kB)
Collecting tokenizers>=0.13.2 (from chromadb<0.4.0,>=0.3.21->langflow)
  Using cached tokenizers-0.20.3-cp313-none-win_amd64.whl.metadata (6.9 kB)
Collecting tqdm>=4.65.0 (from chromadb<0.4.0,>=0.3.21->langflow)
  Using cached tqdm-4.67.0-py3-none-any.whl.metadata (57 kB)
Collecting overrides>=7.3.1 (from chromadb<0.4.0,>=0.3.21->langflow)
  Using cached overrides-7.7.0-py3-none-any.whl.metadata (5.8 kB)
Collecting starlette<0.26.0,>=0.25.0 (from fastapi<0.93.0,>=0.92.0->langflow)
  Using cached starlette-0.25.0-py3-none-any.whl.metadata (5.8 kB)
Collecting httplib2<1.dev0,>=0.19.0 (from google-api-python-client<3.0.0,>=2.79.0->langflow)
  Using cached httplib2-0.22.0-py3-none-any.whl.metadata (2.6 kB)
Collecting google-auth!=2.24.0,!=2.25.0,<3.0.0.dev0,>=1.32.0 (from google-api-python-client<3.0.0,>=2.79.0->langflow)
  Using cached google_auth-2.36.0-py2.py3-none-any.whl.metadata (4.7 kB)
Collecting google-auth-httplib2<1.0.0,>=0.2.0 (from google-api-python-client<3.0.0,>=2.79.0->langflow)
  Using cached google_auth_httplib2-0.2.0-py2.py3-none-any.whl.metadata (2.2 kB)
Collecting google-api-core!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.0,<3.0.0.dev0,>=1.31.5 (from google-api-python-client<3.0.0,>=2.79.0->langflow)
  Using cached google_api_core-2.22.0-py3-none-any.whl.metadata (2.9 kB)
Collecting uritemplate<5,>=3.0.1 (from google-api-python-client<3.0.0,>=2.79.0->langflow)
  Using cached uritemplate-4.1.1-py2.py3-none-any.whl.metadata (2.9 kB)
Collecting cachetools (from gptcache<0.2.0,>=0.1.23->langflow)
  Using cached cachetools-5.5.0-py3-none-any.whl.metadata (5.3 kB)
Requirement already satisfied: setuptools>=3.0 in c:\users\cpolanish\appdata\local\programs\python\python313\lib\site-packages (from gunicorn<21.0.0,>=20.1.0->langflow) (75.3.0)
Collecting filelock (from huggingface-hub<0.14.0,>=0.13.3->langflow)
  Using cached filelock-3.16.1-py3-none-any.whl.metadata (2.9 kB)
Collecting pyyaml>=5.1 (from huggingface-hub<0.14.0,>=0.13.3->langflow)
  Using cached PyYAML-6.0.2-cp313-cp313-win_amd64.whl.metadata (2.1 kB)
Collecting packaging>=20.9 (from huggingface-hub<0.14.0,>=0.13.3->langflow)
  Using cached packaging-24.2-py3-none-any.whl.metadata (3.2 kB)
Collecting SQLAlchemy<3,>=1.4 (from langchain<0.0.177,>=0.0.176->langflow)
  Using cached SQLAlchemy-2.0.36-cp313-cp313-win_amd64.whl.metadata (9.9 kB)
Collecting aiohttp<4.0.0,>=3.8.3 (from langchain<0.0.177,>=0.0.176->langflow)
  Using cached aiohttp-3.10.10-cp313-cp313-win_amd64.whl.metadata (7.8 kB)
Collecting dataclasses-json<0.6.0,>=0.5.7 (from langchain<0.0.177,>=0.0.176->langflow)
  Using cached dataclasses_json-0.5.9-py3-none-any.whl.metadata (22 kB)
Collecting numexpr<3.0.0,>=2.8.4 (from langchain<0.0.177,>=0.0.176->langflow)
  Using cached numexpr-2.10.1.tar.gz (101 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting numpy>=1.21.6 (from chromadb<0.4.0,>=0.3.21->langflow)
  Using cached numpy-1.26.4.tar.gz (15.8 MB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
Collecting openapi-schema-pydantic<2.0,>=1.2 (from langchain<0.0.177,>=0.0.176->langflow)
  Using cached openapi_schema_pydantic-1.2.4-py3-none-any.whl.metadata (8.5 kB)
Collecting tenacity<9.0.0,>=8.1.0 (from langchain<0.0.177,>=0.0.176->langflow)
  Using cached tenacity-8.5.0-py3-none-any.whl.metadata (1.2 kB)
Collecting python-dateutil>=2.8.1 (from pandas<2.0.0,>=1.5.3->langflow)
  Using cached python_dateutil-2.9.0.post0-py2.py3-none-any.whl.metadata (8.4 kB)
Collecting pytz>=2020.1 (from pandas<2.0.0,>=1.5.3->langflow)
  Using cached pytz-2024.2-py2.py3-none-any.whl.metadata (22 kB)
Collecting chardet (from pysrt<2.0.0,>=1.1.2->langflow)
  Using cached chardet-5.2.0-py3-none-any.whl.metadata (3.4 kB)
Collecting markdown-it-py>=2.2.0 (from rich<14.0.0,>=13.3.3->langflow)
  Using cached markdown_it_py-3.0.0-py3-none-any.whl.metadata (6.9 kB)
Collecting pygments<3.0.0,>=2.13.0 (from rich<14.0.0,>=13.3.3->langflow)
  Using cached pygments-2.18.0-py3-none-any.whl.metadata (2.5 kB)
Collecting regex>=2022.1.18 (from tiktoken<0.4.0,>=0.3.3->langflow)
  Using cached regex-2024.11.6-cp313-cp313-win_amd64.whl.metadata (41 kB)
Collecting click<9.0.0,>=7.1.1 (from typer<0.8.0,>=0.7.0->langflow)
  Using cached click-8.1.7-py3-none-any.whl.metadata (3.0 kB)
Collecting argilla (from unstructured<0.6.0,>=0.5.11->langflow)
  Using cached argilla-1.10.0-py3-none-any.whl.metadata (13 kB)
Collecting msg-parser (from unstructured<0.6.0,>=0.5.11->langflow)
  Using cached msg_parser-1.2.0-py2.py3-none-any.whl.metadata (3.0 kB)
Collecting nltk (from unstructured<0.6.0,>=0.5.11->langflow)
  Using cached nltk-3.9.1-py3-none-any.whl.metadata (2.9 kB)
Collecting openpyxl (from unstructured<0.6.0,>=0.5.11->langflow)
  Using cached openpyxl-3.1.5-py2.py3-none-any.whl.metadata (2.5 kB)
Collecting pillow (from unstructured<0.6.0,>=0.5.11->langflow)
  Using cached pillow-11.0.0-cp313-cp313-win_amd64.whl.metadata (9.3 kB)
Collecting pypandoc (from unstructured<0.6.0,>=0.5.11->langflow)
  Using cached pypandoc-1.14-py3-none-any.whl.metadata (16 kB)
Collecting python-docx (from unstructured<0.6.0,>=0.5.11->langflow)
  Using cached python_docx-1.1.2-py3-none-any.whl.metadata (2.0 kB)
Collecting python-pptx (from unstructured<0.6.0,>=0.5.11->langflow)
  Using cached python_pptx-1.0.2-py3-none-any.whl.metadata (2.5 kB)
Collecting python-magic (from unstructured<0.6.0,>=0.5.11->langflow)
  Using cached python_magic-0.4.27-py2.py3-none-any.whl.metadata (5.8 kB)
Collecting markdown (from unstructured<0.6.0,>=0.5.11->langflow)
  Using cached Markdown-3.7-py3-none-any.whl.metadata (7.0 kB)
Collecting certifi>=2022.12.07 (from unstructured<0.6.0,>=0.5.11->langflow)
  Using cached certifi-2024.8.30-py3-none-any.whl.metadata (2.2 kB)
Collecting h11>=0.8 (from uvicorn<0.21.0,>=0.20.0->langflow)
  Using cached h11-0.14.0-py3-none-any.whl.metadata (8.2 kB)
Collecting aiohappyeyeballs>=2.3.0 (from aiohttp<4.0.0,>=3.8.3->langchain<0.0.177,>=0.0.176->langflow)
  Using cached aiohappyeyeballs-2.4.3-py3-none-any.whl.metadata (6.1 kB)
Collecting aiosignal>=1.1.2 (from aiohttp<4.0.0,>=3.8.3->langchain<0.0.177,>=0.0.176->langflow)
  Using cached aiosignal-1.3.1-py3-none-any.whl.metadata (4.0 kB)
Collecting attrs>=17.3.0 (from aiohttp<4.0.0,>=3.8.3->langchain<0.0.177,>=0.0.176->langflow)
  Using cached attrs-24.2.0-py3-none-any.whl.metadata (11 kB)
Collecting frozenlist>=1.1.1 (from aiohttp<4.0.0,>=3.8.3->langchain<0.0.177,>=0.0.176->langflow)
  Using cached frozenlist-1.5.0-cp313-cp313-win_amd64.whl.metadata (14 kB)
Collecting multidict<7.0,>=4.5 (from aiohttp<4.0.0,>=3.8.3->langchain<0.0.177,>=0.0.176->langflow)
  Using cached multidict-6.1.0-cp313-cp313-win_amd64.whl.metadata (5.1 kB)
Collecting yarl<2.0,>=1.12.0 (from aiohttp<4.0.0,>=3.8.3->langchain<0.0.177,>=0.0.176->langflow)
  Using cached yarl-1.17.1-cp313-cp313-win_amd64.whl.metadata (66 kB)
Collecting colorama (from click<9.0.0,>=7.1.1->typer<0.8.0,>=0.7.0->langflow)
  Using cached colorama-0.4.6-py2.py3-none-any.whl.metadata (17 kB)
Collecting urllib3>=1.26 (from clickhouse-connect>=0.5.7->chromadb<0.4.0,>=0.3.21->langflow)
  Using cached urllib3-2.2.3-py3-none-any.whl.metadata (6.5 kB)
Collecting zstandard (from clickhouse-connect>=0.5.7->chromadb<0.4.0,>=0.3.21->langflow)
  Using cached zstandard-0.23.0-cp313-cp313-win_amd64.whl.metadata (3.0 kB)
Collecting lz4 (from clickhouse-connect>=0.5.7->chromadb<0.4.0,>=0.3.21->langflow)
  Using cached lz4-4.3.3.tar.gz (171 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting marshmallow<4.0.0,>=3.3.0 (from dataclasses-json<0.6.0,>=0.5.7->langchain<0.0.177,>=0.0.176->langflow)
  Using cached marshmallow-3.23.1-py3-none-any.whl.metadata (7.5 kB)
Collecting marshmallow-enum<2.0.0,>=1.5.1 (from dataclasses-json<0.6.0,>=0.5.7->langchain<0.0.177,>=0.0.176->langflow)
  Using cached marshmallow_enum-1.5.1-py2.py3-none-any.whl.metadata (2.5 kB)
Collecting typing-inspect>=0.4.0 (from dataclasses-json<0.6.0,>=0.5.7->langchain<0.0.177,>=0.0.176->langflow)
  Using cached typing_inspect-0.9.0-py3-none-any.whl.metadata (1.5 kB)
Collecting googleapis-common-protos<2.0.dev0,>=1.56.2 (from google-api-core!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.0,<3.0.0.dev0,>=1.31.5->google-api-python-client<3.0.0,>=2.79.0->langflow)
  Using cached googleapis_common_protos-1.65.0-py2.py3-none-any.whl.metadata (1.5 kB)
Collecting protobuf!=3.20.0,!=3.20.1,!=4.21.0,!=4.21.1,!=4.21.2,!=4.21.3,!=4.21.4,!=4.21.5,<6.0.0.dev0,>=3.19.5 (from google-api-core!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.0,<3.0.0.dev0,>=1.31.5->google-api-python-client<3.0.0,>=2.79.0->langflow)
  Using cached protobuf-5.28.3-cp310-abi3-win_amd64.whl.metadata (592 bytes)
Collecting proto-plus<2.0.0dev,>=1.22.3 (from google-api-core!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.0,<3.0.0.dev0,>=1.31.5->google-api-python-client<3.0.0,>=2.79.0->langflow)
  Using cached proto_plus-1.25.0-py3-none-any.whl.metadata (2.2 kB)
Collecting pyasn1-modules>=0.2.1 (from google-auth!=2.24.0,!=2.25.0,<3.0.0.dev0,>=1.32.0->google-api-python-client<3.0.0,>=2.79.0->langflow)
  Using cached pyasn1_modules-0.4.1-py3-none-any.whl.metadata (3.5 kB)
Collecting rsa<5,>=3.1.4 (from google-auth!=2.24.0,!=2.25.0,<3.0.0.dev0,>=1.32.0->google-api-python-client<3.0.0,>=2.79.0->langflow)
  Using cached rsa-4.9-py3-none-any.whl.metadata (4.2 kB)
Collecting pyparsing!=3.0.0,!=3.0.1,!=3.0.2,!=3.0.3,<4,>=2.4.2 (from httplib2<1.dev0,>=0.19.0->google-api-python-client<3.0.0,>=2.79.0->langflow)
  Using cached pyparsing-3.2.0-py3-none-any.whl.metadata (5.0 kB)
Collecting mdurl~=0.1 (from markdown-it-py>=2.2.0->rich<14.0.0,>=13.3.3->langflow)
  Using cached mdurl-0.1.2-py3-none-any.whl.metadata (1.6 kB)
Collecting coloredlogs (from onnxruntime>=1.14.1->chromadb<0.4.0,>=0.3.21->langflow)
  Using cached coloredlogs-15.0.1-py2.py3-none-any.whl.metadata (12 kB)
Collecting flatbuffers (from onnxruntime>=1.14.1->chromadb<0.4.0,>=0.3.21->langflow)
  Using cached flatbuffers-24.3.25-py2.py3-none-any.whl.metadata (850 bytes)
Collecting sympy (from onnxruntime>=1.14.1->chromadb<0.4.0,>=0.3.21->langflow)
  Using cached sympy-1.13.3-py3-none-any.whl.metadata (12 kB)
Collecting six>=1.5 (from posthog>=2.4.0->chromadb<0.4.0,>=0.3.21->langflow)
  Using cached six-1.16.0-py2.py3-none-any.whl.metadata (1.8 kB)
Collecting monotonic>=1.5 (from posthog>=2.4.0->chromadb<0.4.0,>=0.3.21->langflow)
  Using cached monotonic-1.6-py2.py3-none-any.whl.metadata (1.5 kB)
Collecting backoff>=1.10.0 (from posthog>=2.4.0->chromadb<0.4.0,>=0.3.21->langflow)
  Using cached backoff-2.2.1-py3-none-any.whl.metadata (14 kB)
Collecting charset-normalizer<4,>=2 (from requests>=2.28->chromadb<0.4.0,>=0.3.21->langflow)
  Using cached charset_normalizer-3.4.0-cp313-cp313-win_amd64.whl.metadata (34 kB)
Collecting idna<4,>=2.5 (from requests>=2.28->chromadb<0.4.0,>=0.3.21->langflow)
  Using cached idna-3.10-py3-none-any.whl.metadata (10 kB)
Collecting anyio<5,>=3.4.0 (from starlette<0.26.0,>=0.25.0->fastapi<0.93.0,>=0.92.0->langflow)
  Using cached anyio-4.6.2.post1-py3-none-any.whl.metadata (4.7 kB)
INFO: pip is looking at multiple versions of tokenizers to determine which version is compatible with other requirements. This could take a while.
Collecting tokenizers>=0.13.2 (from chromadb<0.4.0,>=0.3.21->langflow)
  Using cached tokenizers-0.20.2-cp313-none-win_amd64.whl.metadata (6.9 kB)
  Using cached tokenizers-0.20.1.tar.gz (339 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
  Using cached tokenizers-0.20.0.tar.gz (337 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
  Using cached tokenizers-0.19.1.tar.gz (321 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
  Using cached tokenizers-0.19.0.tar.gz (320 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
  Using cached tokenizers-0.15.2.tar.gz (320 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
  Using cached tokenizers-0.15.1.tar.gz (320 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
INFO: pip is still looking at multiple versions of tokenizers to determine which version is compatible with other requirements. This could take a while.
  Using cached tokenizers-0.15.0.tar.gz (318 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
  Using cached tokenizers-0.14.1.tar.gz (317 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
  Using cached tokenizers-0.14.0.tar.gz (286 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
  Using cached tokenizers-0.13.3.tar.gz (314 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting httptools>=0.5.0 (from uvicorn[standard]>=0.18.3->chromadb<0.4.0,>=0.3.21->langflow)
  Using cached httptools-0.6.4-cp313-cp313-win_amd64.whl.metadata (3.7 kB)
Collecting python-dotenv>=0.13 (from uvicorn[standard]>=0.18.3->chromadb<0.4.0,>=0.3.21->langflow)
  Using cached python_dotenv-1.0.1-py3-none-any.whl.metadata (23 kB)
Collecting watchfiles>=0.13 (from uvicorn[standard]>=0.18.3->chromadb<0.4.0,>=0.3.21->langflow)
  Using cached watchfiles-0.24.0-cp313-none-win_amd64.whl.metadata (5.0 kB)
Collecting httpx<0.24,>=0.15 (from argilla->unstructured<0.6.0,>=0.5.11->langflow)
  Using cached httpx-0.23.3-py3-none-any.whl.metadata (7.1 kB)
Collecting deprecated~=1.2.0 (from argilla->unstructured<0.6.0,>=0.5.11->langflow)
  Using cached Deprecated-1.2.14-py2.py3-none-any.whl.metadata (5.4 kB)
Collecting wrapt<1.15,>=1.13 (from argilla->unstructured<0.6.0,>=0.5.11->langflow)
  Using cached wrapt-1.14.1.tar.gz (50 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting numpy>=1.21.6 (from chromadb<0.4.0,>=0.3.21->langflow)
  Using cached numpy-1.23.5.tar.gz (10.7 MB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error

  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [32 lines of output]
      Traceback (most recent call last):
        File "C:\Users\cpolanish\AppData\Local\Programs\Python\Python313\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 353, in <module>
          main()
          ~~~~^^
        File "C:\Users\cpolanish\AppData\Local\Programs\Python\Python313\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ~~~~^^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\cpolanish\AppData\Local\Programs\Python\Python313\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 112, in get_requires_for_build_wheel
          backend = _build_backend()
        File "C:\Users\cpolanish\AppData\Local\Programs\Python\Python313\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 77, in _build_backend
          obj = import_module(mod_path)
        File "C:\Users\cpolanish\AppData\Local\Programs\Python\Python313\Lib\importlib\__init__.py", line 88, in import_module
          return _bootstrap._gcd_import(name[level:], package, level)
                 ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
        File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
        File "<frozen importlib._bootstrap>", line 1310, in _find_and_load_unlocked
        File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
        File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
        File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
        File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
        File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
        File "<frozen importlib._bootstrap_external>", line 1022, in exec_module
        File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
        File "C:\Users\cpolanish\AppData\Local\Temp\pip-build-env-godnk9se\overlay\Lib\site-packages\setuptools\__init__.py", line 16, in <module>
          import setuptools.version
        File "C:\Users\cpolanish\AppData\Local\Temp\pip-build-env-godnk9se\overlay\Lib\site-packages\setuptools\version.py", line 1, in <module>
          import pkg_resources
        File "C:\Users\cpolanish\AppData\Local\Temp\pip-build-env-godnk9se\overlay\Lib\site-packages\pkg_resources\__init__.py", line 2172, in <module>
          register_finder(pkgutil.ImpImporter, find_on_path)
                          ^^^^^^^^^^^^^^^^^^^
      AttributeError: module 'pkgutil' has no attribute 'ImpImporter'. Did you mean: 'zipimporter'?
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
 cpolanish@FBS-DevAVD-CP2  ~   
```
</details>

If I can provide any other details please let me know! Thanks!

### Reproduction

1. run `python -m pip install langflow -U`
2. Get error

### Expected behavior

Should install succesfully

### Operating System

Windows 11 (running in Azure Virtual Desktop, if it matters)

### Langflow Version

Latest

### Python Version

3.12

**Comment by cpolanish** [comment]
Just wanted to add an update, installing on Python 3.11 worked fine, so seems to be an issue with later version of Python.

**Comment by carlosrcoelho** [comment]
Hey @cpolanish 
We hope to include Python 3.13 in the future. There are some libraries that need adjustments before we can adopt it in Langflow, like this one for example:
https://github.com/facebookresearch/faiss/issues/3985

In the documentation, the Python support issue will be updated.

**Comment by DrAsimKhwaja** [comment]
I am getting dependency errors with both Python 3.10 and 3.12.

**Comment by carlosrcoelho** [comment]
@DrAsimKhwaja 
Please follow the documentation for installation.

https://docs.langflow.org/get-started-installation

If you encounter any problems, please open a new issue so we can assist you. Thank you.
